app/scripts/lib/rpc-method-middleware/handlers/wallet-createSession/handler.ts

```diff
@@ -133,6 +139,24 @@ async function walletCreateSessionHandler(
       }
     };
 
+    // if solana is a requested scope but not supported, we need to add a metadata field to the request
+    const isSolanaRequested =
+      Object.keys(requiredScopesWithSupportedMethodsAndNotifications).includes(
+        MultichainNetwork.Solana,
+      ) ||
+      Object.keys(optionalScopesWithSupportedMethodsAndNotifications).includes(
+        MultichainNetwork.Solana,
+      );
```

Comment:
check for namespace instead?

Reply:
done here: https://github.com/MetaMask/metamask-extension/pull/31544/commits/4b17e871c2cda9fa0153fb3d81e9f087e2d7a6f9#diff-595dd2fe0233f8be65cd7bd3295a5e882fd8ebf623be4a4c0055c8d7278b6444R145-R153